### PR TITLE
Move the bundle to a new namespace

### PR DESCRIPTION
**Pull request by martin-georgiev**

### CODE REVIEW

File: src/DependencyInjection/GoCardlessEnterpriseExtension.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Lendable\GoCardlessEnterpriseBundle\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\HttpKernel\DependencyInjection\Extension as SymfonyExtension;
+use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+
+class GoCardlessEnterpriseExtension extends SymfonyExtension
```
Comment (martin-georgiev) [inline src/DependencyInjection/GoCardlessEnterpriseExtension.php]: There are only namespace and class name case changes in this file, however, GitHub fails short of displaying this nicely and thinks the file is new.

File: src/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -1,36 +1,29 @@
...
     public function getConfigTreeBuilder()
     {
-        $treeBuilder = new TreeBuilder();
-        $rootNode = $treeBuilder->root('gocardless_enterprise');
+        $treeBuilder = new TreeBuilder('gocardless_enterprise');
+        if (method_exists($treeBuilder, 'getRootNode')) {
+            $rootNode = $treeBuilder->getRootNode();
+        } else {
+            $rootNode = $treeBuilder->root('gocardless_enterprise');
+        }
```
Comment (martin-georgiev) [inline src/DependencyInjection/Configuration.php]: Above lines were changed to support the [newer Symfony approach for TreeBuilder](https://symfony.com/blog/new-in-symfony-4-2-important-deprecations#deprecated-tree-builders-without-root-nodes).

File: composer.json [hunk 3]
```diff
@@ -14,12 +14,12 @@
   
   "autoload": {
     "psr-4": {
-      "Gocardless\\EnterpriseBundle\\": "src/"
+      "Lendable\\GoCardlessEnterpriseBundle\\": "src/"
```
Comment (martin-georgiev) [inline composer.json]: Namespace changed so it is in line with the new namespace of gocardless-enterprise (that got [merged in summer 2018](https://github.com/Lendable/gocardless-enterprise/pull/9/files) but wasn't in use by the bundle until now). The new namespace also kills the case variation in `GoCardless` that existed before.

File: .travis.yml [hunk 4]
```diff
@@ -78,3 +78,7 @@ script:
     else
       composer check-security;
     fi;
+  - composer run-tests-with-clover
```
Comment (martin-georgiev) [inline .travis.yml]: Tests now will run against all Travis matricies.

File: src/DependencyInjection/Configuration.php [hunk 5]
```diff
@@ -1,36 +1,29 @@
...
-        // Here you should define the parameters that are allowed to
-        // configure your bundle. See the documentation linked above for
-        // more information on that topic.
+        $rootNode
+            ->children()
+            ->scalarNode('baseUrl')->isRequired()->cannotBeEmpty()->end()
+            ->scalarNode('gocardlessVersion')->isRequired()->cannotBeEmpty()->end()
+            ->scalarNode('webhook_secret')->isRequired()->cannotBeEmpty()->end()
+            ->scalarNode('token')->isRequired()->cannotBeEmpty()->end()
+            ->end();
```
Comment (martin-georgiev) [inline src/DependencyInjection/Configuration.php]: It's yet unclear to me why this was left commented out before.

File: tests/DependencyInjection/GoCardlessEnterpriseExtensionTest.php [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+declare(strict_types=1);
+
+namespace Lendable\GoCardlessEnterpriseBundle\Tests\DependencyInjection;
+
+use GuzzleHttp\Client as GuzzleClient;
+use Lendable\GoCardlessEnterprise\Client;
+use Lendable\GoCardlessEnterpriseBundle\DependencyInjection\GoCardlessEnterpriseExtension;
+use Matthias\SymfonyDependencyInjectionTest\PhpUnit\AbstractExtensionTestCase;
+
+class GoCardlessEnterpriseExtensionTest extends AbstractExtensionTestCase
```
Comment (martin-georgiev) [inline tests/DependencyInjection/GoCardlessEnterpriseExtensionTest.php]: This test is new.

File: src/DependencyInjection/Configuration.php [hunk 7]
```diff
@@ -1,36 +1,29 @@
...
 {
-    /**
-     * {@inheritDoc}
-     */
     public function getConfigTreeBuilder()
     {
-        $treeBuilder = new TreeBuilder();
-        $rootNode = $treeBuilder->root('gocardless_enterprise');
+        $treeBuilder = new TreeBuilder('gocardless_enterprise');
+        if (method_exists($treeBuilder, 'getRootNode')) {
```
Comment (robwasripped) [inline src/DependencyInjection/Configuration.php]: nit: This could be a ternary to avoid a branching statement

**Comment by martin-georgiev** [comment]
> The entities could benefit from a `use` statement each for the model they're extending.

Note that by doing so all `use` statements will have to be aliased (because of the name conflict). Therefore I'm not convinced it will improve the readability.

File: src/DependencyInjection/Configuration.php (hunk 7 above)
Comment (martin-georgiev) [reply src/DependencyInjection/Configuration.php]: True, however, readability won't improve imo as the check and two different calls are somewhat long. The ternary will need splitting across multiple lines.

**Pull request merged**